### ISSUE TITLE
NullReferenceException in FieldAccessAnalyzer

### ISSUE THREAD

**Issue by TheXenocide** [icsharpcode/ILSpy#1376]
Version: 4.0.0.4319-beta2

While trying to analyze usage of various fields in HttpClientHandler (e.g. `System.Net.Http.HttpClientHandler+RequestState.requestMessage`) I get:

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at ICSharpCode.ILSpy.Analyzers.Builtin.FieldAccessAnalyzer.<Analyze>d__4.MoveNext()
   at ICSharpCode.ILSpy.Analyzers.AnalyzerSearchTreeNode.<FetchChildren>d__10.MoveNext()
   at ICSharpCode.ILSpy.TreeNodes.ThreadingSupport.<>c__DisplayClass5_0.<LoadChildren>b__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
```

**Comment by TheXenocide** [comment]
Actually, interestingly I'm now getting NullReferenceException from all analyzers. What's curious is that they were working before. I tried closing and re-opening ILSpy and am still getting them. Here's another stack trace:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at ICSharpCode.ILSpy.Analyzers.Builtin.TypeInstantiatedByAnalyzer.<Analyze>d__1.MoveNext()
   at ICSharpCode.ILSpy.Analyzers.AnalyzerSearchTreeNode.<FetchChildren>d__10.MoveNext()
   at ICSharpCode.ILSpy.TreeNodes.ThreadingSupport.<>c__DisplayClass5_0.<LoadChildren>b__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
```
I've captured a memory dump, in case that helps, but am unable to attach here due to size. I'll hold onto it for a bit so let me know if you need it.

**Comment by siegfriedpammer** [comment]
The results of analysis depend on the assemblies you have added to the current list. I am not sure how we could debug this issue without knowing which assembly causes the analyzer to fail. I could take a look at the memory dump... please send it to siegfried.pammer@gmail.com. Thank you!

**Comment by siegfriedpammer** [comment]
Any update on this?

**Comment by TheXenocide** [comment]
Apologies for the delay. Initially I didn't see the notification for your previous message, but I have just shared the file with you from Google Drive. Unfortunately, I'm no longer able to reproduce the issue (at some point I think removing all inspected assemblies, closing, reopening, and re-decompiling seemed to get me past the issue). I've since moved on to other work, but if there's anything I can do to help out (understanding I have limited time available), please let me know. I do hope the information helps and thanks for what you guys do!

**Comment by siegfriedpammer** [comment]
Thank you for sending the memory dump. Unfortunately, the relevant exception/error info is missing. However, there's good news: Just now, I was able to reproduce the exception (or a similar exception) by playing around with the beta2. Seems this was fixed in 51ab9a7b99a43ba387d82855860891335d986a76 (beta3 or newer).

Thank you for reporting this anyway! Feedback is always welcome!

>  thanks for what you guys do!

Thank you for using ILSpy!
